eth/blockwatch/block_watcher.go

```diff
@@ -395,8 +416,6 @@ func (w *Watcher) getMissedEventsToBackfill(ctx context.Context) ([]*Event, erro
 					Number: big.NewInt(0).SetUint64(log.BlockNumber),
 					Logs:   []types.Log{},
 				}
-				// Logs don't contain L1 block number, so we need to fetch it separately
-				blockHeader, err = w.enrichWithL1BlockNumber(blockHeader)
```

Comment:
If we are backfilling don't we still need this so that the events generated from backfilling have L1 block numbers attached to the headers?

Reply:
This is no need to do it here anymore since we moved it to `enrichWithL1()` and enrich only the highest block number [here](https://github.com/leszko/go-livepeer/blob/1382f59ad78aa7f819ba84207d1d1ad4c90e5e6b/eth/blockwatch/block_watcher.go#L100)